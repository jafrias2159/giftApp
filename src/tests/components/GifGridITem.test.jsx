import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import React from "react";

describe("Tests on GifGridItem", () => {
  const title = "saitama";
  const url = "japan.anime/";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Should render a gifGridItem", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should render a p element", () => {
    const pElement = wrapper.find("p");
    expect(pElement.text().trim()).toBe(title);
  });

  test("should render an img with a src and alt props", () => {
    const imgElement = wrapper.find("img");

    expect(imgElement.prop("src")).toBe(url);
  });

  test('the main container must have the "animate__fadeIn" class', () => {
    const divElement =  wrapper.find("div");
    const animateClass = "animate__fadeIn";
    expect(divElement.prop('className')).toContain(animateClass);
  });
});
