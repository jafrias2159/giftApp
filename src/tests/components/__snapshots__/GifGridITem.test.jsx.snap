// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItem Should render a gifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="saitama"
    src="japan.anime/"
  />
  <p>
     
    saitama
     
  </p>
</div>
`;
